content/helpers.go

```diff
@@ -28,7 +29,10 @@ func NewReader(ra ReaderAt) io.Reader {
 //
 // Avoid using this for large blobs, such as layers.
 func ReadBlob(ctx context.Context, provider Provider, dgst digest.Digest) ([]byte, error) {
```

Comment:
This should take the descriptor as a parameter instead. All cases that I found where this was called it already took a digest from a descriptor before calling.